src/sentry/incidents/logic.py

```diff
@@ -270,24 +270,29 @@ def _unpack_organization(alert_rule: AlertRule) -> Organization:
     return organization
 
 
-def _build_incident_query_builder(
-    incident: Incident,
+def _build_metric_query_builder(
+    snuba_query: SnubaQuery,
+    organization: Organization,
+    project_ids: list[int],
```

Comment:
Does it make sense to combine these into a dataclass somehow that we could simplify passing around/converting from an incident later? It might shorten the function call sizes for these and make them a little less typo prone.